CMakeLists.txt

```
@@ -33,6 +33,8 @@ option(PORTABLE "build a portable binary (disable arch-specific optimizations)"
 # TODO: set ENABLE_NEW_ENCODING to ON when we are ready
 option(ENABLE_NEW_ENCODING "enable new encoding (#1033) for storing 64bit size and expire time in milliseconds" OFF)
 
+set(CMAKE_LINK_DEPENDS_NO_SHARED 1)
```

```suggestion
set(CMAKE_LINK_DEPENDS_NO_SHARED ON)
```

And, could you share some reason for this change?

---

just avoid relinking all depended targets on .so. It't necessary now, so I delete it.